### PR TITLE
[Algorithm] Support  Oriented Reppoints (CVPR'22)

### PR DESCRIPTION
**Pull request by LiWentomng**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Support the [Oriented RepPoints](https://arxiv.org/pdf/2105.11111v4.pdf) (CVPR'22)  for  [#127](https://github.com/open-mmlab/mmrotate/issues/127)

## Modification

-  Add new head:  [oriented_reppoints_head](https://github.com/LiWentomng/mmrotate/blob/main/mmrotate/models/dense_heads/oriented_reppoints_head.py)
- Add new loss:  [spatial_border_loss](https://github.com/LiWentomng/mmrotate/blob/main/mmrotate/models/losses/spatial_border_loss.py)
- Add new basic config: [oriented_reppoints_r50_fpn_1x_dota_le135](https://github.com/LiWentomng/mmrotate/blob/main/configs/oriented_reppoints/oriented_reppoints_r50_fpn_1x_dota_le135.py),  [config_README](https://github.com/LiWentomng/mmrotate/blob/main/configs/oriented_reppoints/README.md)

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by yangxue0827** [comment]
- Docstring is deeded in [oriented_reppoints_head](https://github.com/LiWentomng/mmrotate/blob/main/mmrotate/models/dense_heads/oriented_reppoints_head.py) and [spatial_border_loss](https://github.com/LiWentomng/mmrotate/blob/main/mmrotate/models/losses/spatial_border_loss.py).
- Support algorithms in [README.md](https://github.com/open-mmlab/mmrotate/blob/main/README.md) and [README_zh-CN.md](https://github.com/open-mmlab/mmrotate/blob/main/README_zh-CN.md) should be updated.

**Comment by yangxue0827** [comment]
It would be nice to add a docstring for each function.

**Comment by jbwang1997** [comment]
Hello @LiWentomng. 

Thanks for your contributions. Seems the unit test cannot pass. Please remember to check. There are also some lint errors. You can refer to https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md to install pre-commit to fix the lint error.

**Comment by zytx121** [comment]
Waiting for the chamber distance in mmcv.

**Comment by LiWentomng** [comment]
Thanks a lot.  I will focus on this to make it.   I update the `oriented_reppoints_head.py` and add a new config.

File: README.md [hunk 1]
```diff
@@ -64,9 +64,9 @@ https://user-images.githubusercontent.com/10410257/154433305-416d129b-60c8-44c7-
 
 ## Changelog
```
Comment (yangxue0827) [inline README.md:65]: The changelog does not need to be modified, it will be modified by the maintainer when a new version is released.

**Comment by yangxue0827** [comment]
Waiting for train log and trained weights.

Do we need to add code to check the mmcv version. @zytx121

**Comment by zytx121** [comment]
> Waiting for training log and trained weights.
> 
> Do we need to add code to check the mmcv version. @zytx121

I think we can just bump the min mmcv version.

**Comment by zytx121** [comment]
Please add your models in these docs:
https://github.com/open-mmlab/mmrotate/blob/main/docs/zh_cn/model_zoo.md
https://github.com/open-mmlab/mmrotate/blob/main/docs/en/model_zoo.md

**Comment by yangxue0827** [comment]
This branch has conflicts that must be resolved

**Review by yangxue0827 — APPROVED** [review]
LGTM

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @LiWentomng ！First of all, we want to express our gratitude for your significant PR in the mmrotate project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**